### PR TITLE
feat: add prettier tip

### LINKED ISSUE #5: New tip: setup prettier
- also tailwindcss, order matters

### PR DESCRIPTION
**Pull request by at-the-vr**
- closes #5 
- gives a brief rundown for installing and configuring Prettier with Astro Projects
- covers tailwindcss plugin for class formatting

### CODE REVIEW

File: src/content/docs/tips/prettier.mdx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+---
+title: Setting up Prettier
+description: Quickly configure Prettier to your Astro Project
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:3]:
```suggestion
description: Quickly configure Prettier for your Astro project
```

File: src/content/docs/tips/prettier.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+---
+import { Tabs, TabItem, LinkCard } from "@astrojs/starlight/components";
+
+## Prettier Plugin for Astro
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:7]:
```suggestion
Quickly configure Prettier for your Astro project!

## Prettier Plugin for Astro
```

File: src/content/docs/tips/prettier.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        ```
+    </TabItem>
+    <TabItem label="yarn">
+        ```sh
+        yarn add --dev prettier prettier-plugin-astro
+        ```
+    </TabItem>
+</Tabs>
+
+Next, update [Prettier Config](https://prettier.io/docs/en/configuration) in the project's root to include `prettier-plugin-astro` plugin. This example will use `.prettierrc`:
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:29]:
```suggestion
Next, update your [Prettier config](https://prettier.io/docs/en/configuration) in the project's root to include `prettier-plugin-astro` plugin. This example will use `.prettierrc.json`:
```

File: src/content/docs/tips/prettier.mdx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+```json title=".prettierrc"
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:31]:
```suggestion
```json title=".prettierrc.json"
```

File: src/content/docs/tips/prettier.mdx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    }
+  ]
+}
+```
+
+That's it. Run `prettier --write ./src` command with your preferred package manager to format all of the astro files in the project.
+
+Additionally if you are using the [VSCode Prettier Extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode), you can set the following VSCode Config in your User/Workspace Settings:
+
+```json title="settings.json"
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:49]:
```suggestion
```json title=".vscode/settings.json"
```

File: src/content/docs/tips/prettier.mdx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  "prettier.documentSelectors": ["**/*.astro"],
+  "[astro]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+  }
+}
+```
+
+and now Prettier will format any selected document.
+
+## Prettier Plugin for TailwindCSS
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:60]:
```suggestion
## Prettier Plugin for Tailwind CSS
```

File: src/content/docs/tips/prettier.mdx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+With TailwindCSS involved there is another dev dependency for automated class sorting that is [prettier-plugin-tailwindcss](https://github.com/tailwindlabs/prettier-plugin-tailwindcss).
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:62]:
```suggestion
When using the Tailwind CSS integration, there is another dev dependency for automated class sorting: [prettier-plugin-tailwindcss](https://github.com/tailwindlabs/prettier-plugin-tailwindcss).
```

File: src/content/docs/tips/prettier.mdx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        yarn add --dev prettier-plugin-tailwindcss
+        ```
+    </TabItem>
+</Tabs>
+
+Next, update the [Prettier Config](https://prettier.io/docs/en/configuration) to include following plugins. This example will use `.prettierrc`:
+
+```json title=".prettierrc" "prettier-plugin-tailwindcss"
+{
+  "plugins": ["prettier-plugin-astro","prettier-plugin-tailwindcss"],
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:88]:
```suggestion
```json title=".prettierrc.json" del={2} ins={3}
{
  "plugins": ["prettier-plugin-astro"],
  "plugins": ["prettier-plugin-astro", "prettier-plugin-tailwindcss"],
```

File: src/content/docs/tips/prettier.mdx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+      "files": "*.astro",
+      "options": {
+        "parser": "astro",
+      }
+    }
+  ]
+}
+```
+
+Its important that the tailwindcss plugin comes in the end for Prettier to take necessary formatting. Additionally, you might need to reload your [VSCode Window](https://stackoverflow.com/a/42002990) in the case that Prettier is not working as expected.
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:100]:
```suggestion
Make sure to place the Tailwind CSS plugin at the end. Additionally, you might need to reload your [VSCode Window](https://stackoverflow.com/a/42002990) if Prettier is not working as expected.
```

File: src/content/docs/tips/prettier.mdx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+      "files": "*.astro",
+      "options": {
+        "parser": "astro",
+      }
+    }
+  ]
+}
+```
+
+That's it. Run `prettier --write ./src` command with your preferred package manager to format all of the astro files in the project.
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:45]:
```suggestion
That's it. Run `prettier --write .` command with your preferred package manager to format all of the astro files in the project.
```

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]
Amazing thank you! Can you add a section about adding a package.json script for convenience?

**Comment by at-the-vr** [comment]
> Amazing thank you! Can you add a section about adding a package.json script for convenience?

thanks for the suggestions 🙌 , have a look at it and lemme know if it can be improved 😄

File: src/content/docs/tips/prettier.mdx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+You can use Prettier CLI with your preferred package manager by adding following scripts in `package.json` of your project's directory:
+
+```json title=".package.json" ins={3,4}
+{
+  "scripts": {
+    "format:check": "prettier --check ./src/",
+    "format::write": "prettier --write ./src/"
+  }
+}
+```
```
Comment (florian-lefebvre) [inline src/content/docs/tips/prettier.mdx:117]:
```suggestion
## Adding an NPM script

You can use Prettier CLI with your preferred package manager by adding following scripts in `package.json` of your project's directory:

```json title=".package.json" ins={3,4}
{
  "scripts": {
    "format:check": "prettier --check .",
    "format::write": "prettier --write ."
  }
}
```
```
Also could be great to move this above Tailwind  CSS!

**Review by florian-lefebvre — APPROVED** [review]

**Pull request merged**